### ISSUE TITLE
Toxcore needs a red team

### ISSUE THREAD

**Issue by GrayHatter** [TokTok/c-toxcore#211]

**Comment by sudden6** [comment]
What's a red team and what does it do?

**Comment by dvor** [comment]
> A red team is an independent group that challenges an organization to improve its effectiveness. 

> When used in a hacking context, a red team is a group of white-hat hackers that attack an organization's digital infrastructure as an attacker would in order to test the organization's defenses 

https://en.wikipedia.org/wiki/Red_team

**Comment by isotoxin** [comment]
Ha! You guys just do nothing and this is main problem. Red team...

**Comment by Diadlo** [comment]
@isotoxin Are you sure?
https://github.com/TokTok/c-toxcore/pulse/monthly

**Comment by Renha** [comment]
ha, look, it's mr. @isotoxin aka 'Updated 21 days ago' by himself

**Comment by isotoxin** [comment]
No one bug was fixed last year.
No one optimization was done last year.
No one new feature was implemented last year.

Toxcore! Most stable code ever!

**Comment by Diadlo** [comment]
@isotoxin If you read toktok [plan](http://toktok.github.io/plan.html) on toxcore (sorry for previous broken link. Updated now) there is no bugfixes. But, as it appears, you related with coding => you should understand, that tests, documentation, code cleanup and refactoring make notorious bugfixes easier in the future.

**Comment by GrayHatter** [comment]
I'm going to attempt to slow this flamebate down a bit by locking to collaborators for now.

This is a meta issue and in no way the main problem. (I'm not sure why you think than @isotoxin unless you're just fusterated, which I can understand; Tox has been somewhat slow in the past) I created this issue in case someone else wants to take it on while I work on tox-named, as well as upgrading net_crypto code.

Once that's done I'll take another look at this myself, meanwhile I can't forget about it. The reason I opened this is because I'm planning to change how friends find each other, so we can drastically reduce data usage. While that's VERY important I also don't want to make tox less secure. Having a red team would be a step in the right direction to making AND KEEPING Tox secure.
